src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE_ARRAY = 10;
+    private int size = 0;
+    private final K[] keys;
+    private final V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE_ARRAY];
+        values = (V[]) new Object[MAX_SIZE_ARRAY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            for (int i = 0; i < size; i++) {
+                if ((keys[i] == key) || (keys[i] != null && keys[i].equals(key))) {
+                    values[i] = value;
+                }
+            }
+            return;
+        }
```

Comment:
move return iside if block
```suggestion
        if (get(key) != null) {
            for (int i = 0; i < size; i++) {
                if ((keys[i] == key) || (keys[i] != null && keys[i].equals(key))) {
                    values[i] = value;
                    return;
                }
            }
        }
```